src/config/loading/secret.rs

```diff
@@ -249,6 +249,11 @@ async fn query_backend(
 
     let output = child.wait_with_output().await?;
     let response = serde_json::from_slice::<HashMap<String, ExecResponse>>(&output.stdout)?;
+    let stderr = String::from_utf8_lossy(&output.stderr);
+
+    if !stderr.trim().is_empty() {
+        warn!("A secret exec backend wrote a message on stderr: {}.", stderr);
+    }
```

Comment:
I think this won't read very well for multi-line stderr.

I was also thinking we'd proxy it through _as_ it was written which could aid with troubleshooting if the task is hanging, for example. I'd say we could just do `command.stderr(Stderr::inherit())`, but maybe it would be better to feed it through Vector's log system so it is in a consistent format.  That could mean something like:

* Rather than calling `wait_with_output` call `spawn` on `command` so that you can read the stderr output as it is written
* Call `warn!` with each line of stderr output so that it ends up in Vector's logs with the appropriate format
* Collect the output asynchronously until the command finishes (I think serde can actually take a async reader too so maybe you could just feed it into that)

I'm happy to just defer this for now though, if you prefer, and open a follow up ticket. That would mean dropping the `stderr` handling you have here (maybe just doing `command.stderr(std::process::Stderr::piped())`.

Reply:
I'm also realizing the above:

```
    command.stderr(std::process::Stdio::piped());
    command.stdin(std::process::Stdio::piped());
    command.stdout(std::process::Stdio::piped());
```

Isn't right, they should be piped to `Stderr` `Stdout` `Stdin` respectively.

Reply:
> I'm also realizing the above:
> 
> ```
>     command.stderr(std::process::Stdio::piped());
>     command.stdin(std::process::Stdio::piped());
>     command.stdout(std::process::Stdio::piped());
> ```
> 
> Isn't right, they should be piped to `Stderr` `Stdout` `Stdin` respectively.

This is confusing but the [`Stdio::piped()`](https://doc.rust-lang.org/stable/src/std/process.rs.html#1122-1219) just instructs what to do with stderr/stdout/stdin from the child process and in that case it establishes a pipe between Vector and the child process to read/write to/from stdin/stdio/stderr.

Reply:
Doh, yeah, I misread `Stdio` as `Stdout` 👍